Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,288 @@
+        }
+
+        menuPropDisplay.transform.parent = boundingScaleBox.transform;
+        menuPropDisplay.transform.localPosition = new Vector3(0, 0, 0);
+
+        boundingScaleBox.transform.parent = propTile.transform;
+        boundingScaleBox.transform.localScale = new Vector3(.6f, .6f, .6f);
+
+        if (menuPropDisplay.GetComponent<Collider>())
+        {
+            menuPropDisplay.GetComponent<Collider>().enabled = false;
+        }
+
+        menuPropDisplay.AddComponent<slowRotation>();
+        menuPropDisplay.SetActive(true);
+    }
+
+    void triggerClicked(object sender, ClickedEventArgs e)
+    {
+        Ray ray = new Ray(inputManager.hand2.gameObject.transform.position, inputManager.hand2.gameObject.transform.forward);
+        RaycastHit hit;
+        if (Physics.Raycast(ray, out hit, 100))
+        {
+
+            if (hit.collider.transform.parent == propTiles.transform)
+            {
+                //get the correct object from the current chunk
+                GameObject newProp = Instantiate(chunkedList[currentPage][hit.transform.GetSiblingIndex()]);
+                newProp.name = chunkedList[currentPage][hit.transform.GetSiblingIndex()].name;
+                newProp.transform.parent = propParent.transform;
+                newProp.SetActive(true);
+
+                //set the spawned object to the new selected object
+                _stateManagerMutatorRef.SET_SELECTED_OBJECT(newProp);
+                newProp.transform.position = hit.point;
+                //check for prop -some objects are loaded with props, some aren't. 
+                if(!newProp.GetComponent<prop>())
+                {
+                    newProp.AddComponent<prop>();
+                }
+                newProp.AddComponent<activeProp>();
+                _stateManagerMutatorRef.SET_SELECTED_OBJECT_IS_ACTIVE(true);
+
+            }
+
+            //page incrementer and decrementer
+            if (hit.collider.gameObject.name == "pageUp")
+            {
+                if (currentPage >= chunkedList.Count - 1)
+                {
+                    return;
+                }
+                else
+                {
+                    currentPage += 1;
+                    showPropObjectPage(currentPage);
+                }
+            }
+            if (hit.collider.gameObject.name == "pageDown")
+            {
+                if (currentPage <= 0)
+                {
+                    return;
+                }
+                else
+                {
+                    currentPage -= 1;
+                    showPropObjectPage(currentPage);
+                }
+            }
+            //removes the menu
+            if (hit.collider.gameObject.name == "closeMenu")
+            {
+                toggleMenuComponents(false);
+            }
+        }
+    }
+
+    void toggleMenuComponents(bool on)
+    {
+        foreach (Transform child in transform.parent.GetChild(0))
```

Comment:
what is getChild(0) and getChild(1) referring to?

Reply:
menu is the parent of the sub menus: loadPropsMenu and propertiesMenu. there is propbably a better way to toggle them off...

Reply:
maybe make a [serializablefield] property for them and pass them in explicitly.. just don't want to break stuff like that if things get moved around at all.

Reply:
what about a foreach child loop, that then loops over the grand children to disable them. I know nested for loops are bad. but there will never be a crazy amount of menu items.

Reply:
why cant we just set the parents active? - what's the point of looping through the children?

Reply:
parents are holding classes that need to start updated. so they currently must always be listening

Reply:
i think its time then to just create a menu class that turns on/off different menus depending on the state